src/lib/selections.ts

```
@@ -1089,17 +1089,18 @@ export async function selectionBodyFace(
     )
   }
 
-  const lastChild =
-    findAllChildrenAndOrderByPlaceInCode(
-      { type: 'sweep', ...extrusion },
-      kclManager.artifactGraph
-    )[0] || null
-  const lastChildCodeRef: PathToNode | null =
-    lastChild?.type === 'compositeSolid' ? lastChild.codeRef.pathToNode : null
-
-  const extrudePathToNode = !err(extrusion)
-    ? lastChildCodeRef || extrusion.codeRef.pathToNode
-    : []
+  const children = findAllChildrenAndOrderByPlaceInCode(
+    { type: 'sweep', ...extrusion },
+    kclManager.artifactGraph
+  )
+  const lastChildVariable = getLastVariable(children, kclManager.ast, [
+    'sweep',
+    'compositeSolid',
+  ])
+  const extrudePathToNode =
+    lastChildVariable && !err(lastChildVariable)
+      ? lastChildVariable.pathToNode
+      : []
```

can this fallback cause a silent fail?

---

Good to wait for @Irev-Dev to chime in